src/Core/System/DependencyInjection/sales_channel.xml

```diff
@@ -135,7 +135,9 @@
             <argument type="service" id="event_dispatcher"/>
         </service>
 
-        <service id="Shopware\Core\System\SalesChannel\SalesChannel\ContextRoute"/>
+        <service id="Shopware\Core\System\SalesChannel\SalesChannel\ContextRoute">
```

Comment:
Until now we use `public="true"` for all controllers (Admin-API, Store-API routes and storefront controllers), for the sake of consistency i would suggest to also make this service public and don't use the service_arguments tag

Or do you know any reason why the tag is preferable over making the service public? If there is a good reason we should consider switching to the tag for all controllers

Reply:
@keulinho Symfony's autoconfigure if you use `AsController`  attribute or `extends AbstractController` adds the tag.

Futhermore, the tag makes the service public automatically and it allows service dependency injection in the route's method as well. Though for Store API services, that probably will not be useful as they are used outside of routes as well. However, that feature does not have to be used.

So I can also just make them public.

See: https://symfony.com/doc/current/controller/service.html#how-to-define-controllers-as-services
`Alternatively, you can apply the controller.service_arguments tag to your controller services. This will make the tagged services public and will allow you to inject services in method parameters:`

and autoconfigure:
https://github.com/symfony/symfony/blob/v5.4.15/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L615
https://github.com/symfony/symfony/blob/v5.4.15/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L544

I am fine with using only public as well. but to be more in sync with Symfony's autoconfigure, I think the tag might make more sense.

Reply:
Ok thanks for the explanation :+1: 

i already created a internal ticket that we create a custom phpstan rule to check that all controllers(Admin API, Store-API and Storefront) are correctly configured, to prevent issues like that in the future. 

I would suggest that we use `public="true"` here in this case, but will switch to the  `controller.service_arguments` for all controllers when we add the phpstan rule for that, to be more in line with the symfony default behaviour

The reason is that with that phpstan rule in place it is easier to enforce consistency

Reply:
yeah change it to public here and then we will switch for all controllers to tags in the future ;)

Reply:
@keulinho Done.